### ISSUE TITLE
Firefox showing url bar

### ISSUE THREAD

**Issue by mambucodev** [linuxmint/webapp-manager#320]
since Firefox v133 I am getting a lot of firefox compatibility issues, webapp-manager included.
Now the URL bar just appeared and can't be removed, and if I hover it, the tabs appear too.
This is what I see since I updated firefox:

![image](https://github.com/user-attachments/assets/157a6db0-8a7a-486e-bc20-b81a2ea5a988)

And these are the settings:

![image](https://github.com/user-attachments/assets/99a18629-c425-4ecc-a894-607fb858b614)

**Comment by contrarybaton60** [comment]
glad I'm not the only one having this issue

**Comment by TL2-24** [comment]
Same problem. I also tried creating a new webapp, which showed the same problem.

**Comment by jaymecamargo** [comment]
Same here... :/

**Comment by cordlord** [comment]
Just noticed this too. I did notice Firefox just updated today right before I saw the issue.

I'm guessing there's a correlation there.

**Comment by mambucodev** [comment]
> Just noticed this too. Wasn't there until after I updated to Firefox version 133 as well.

Yes, updating to v133 changed a lot of UI aspects. I also had a GTK theme on firefox and it broke too.

**Comment by alextpedro** [comment]
Same here. It's pretty annoying because moving the mouse over it shows the tabs UI.

**Comment by mambucodev** [comment]
I recently noticed that the gtk theme seems to be already adapted to v133 so I don't understand why addwater isn't updating the theme. This is not certain and needs further investigation.

**Comment by mateimm** [comment]
I believe my issue is related. I use Firefox with navigation bar hidden in Linux Web Apps (in Linux Mint > Web Apps :: where you can create apps from any web page). I particularly use Notion as an app. And recently, I guess with the new updates, there is showing an address bar where previously there was nothing. Is there a way to remove it?
![ksnip_20241129-123813](https://github.com/user-attachments/assets/fae55ef4-ea11-4537-90b7-d7b73aacd36f)
![ksnip_20241129-122850](https://github.com/user-attachments/assets/cf1e26d0-87f6-4a82-9ffe-c3eac43ce555)

**Comment by mambucodev** [comment]
@mateimm yes it's 100% related. a firefox update caused it. I think they're fixing it.

**Comment by contrarybaton60** [comment]
In the meantime you can use the ESR version of FIrefox (either through Mozilla's debian repo or `firefox-esr` package if you're using LMDE)

**Comment by NonRidiculousAdjective** [comment]
Same problem here

**Comment by bionicle19** [comment]
I have the same issue.

**Comment by epikaigle444** [comment]
Same problem here

**Comment by rmburg** [comment]
Please refrain from posting "I have the same problem" comments. Everyone subscribed to this thread receives an e-mail for each comment. Instead, you can give the original issue a thumbs up or similar to show that you are also affected.

**Comment by mambucodev** [comment]
> Please refrain from posting "I have the same problem" comments. Everyone subscribed to this thread receives an e-mail for each comment. Instead, you can give the original issue a thumbs up or similar to show that you are also affected.

Thanks, I was thinking of saying that.
I'd also add something: If you're new to GitHub or simply don't know it, you can subscribe to a issue/thread without the need of commenting. We already know the issue exists and what caused it, so there's no need to tell us.

**Comment by mateimm** [comment]
> this is what i did for now for every `chrome/userChrome.css` under my apps in `.local/share/ice/firefox/`:
> 
> ```css
> #navigator-toolbox {
>     border-bottom: none !important;
> }
> 
> #nav-bar, #identity-box, #tabbrowser-tabs, #TabsToolbar, #nav-bar * {
>     visibility: collapse !important;
> }
> ```
> 
> (you can do it once and copy the file around of course, or use some bash magic)
> 
> it will always hide the bars but i never need them in my web apps.
> 
> i also (hot-patched) it for future web apps under `/usr/share/webapp-manager/firefox/profile/chrome/userChrome.css`
> 
> it works beautifully for my needs.
> 
> ![image](https://private-user-images.githubusercontent.com/1176779/391925483-eb60958c-9ac2-4bf3-b704-06d3a37f9413.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMyNDE3NzQsIm5iZiI6MTczMzI0MTQ3NCwicGF0aCI6Ii8xMTc2Nzc5LzM5MTkyNTQ4My1lYjYwOTU4Yy05YWMyLTRiZjMtYjcwNC0wNmQzYTM3Zjk0MTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDNUMTU1NzU0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MWI0ODU5NDU0MGQ5NTZiYWViMzI0ODUwMDA1Njg3ZDY0YWQ4ODMyYTAzYzBjMDc0NTFiZWMyNThjZTA5NTMzZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.9jlFi8o563WkemOf3QRDSo-t0M2SC9SEHxZgigsUTas)

I have found a workaround. Replacing the CSS code  in  `usr/share/webapp-manager/firefox/profile/chrome/userChrome.css` with the one above and re-creating all the apps is actually removing the navbar for all newely created apps. I guess is the same as adding the code in each `userChrome.css` as mentioned above, but at least if you create new apps you don't have to add the CSS manually if it has already been added to  `usr/share/webapp-manager/firefox/profile/chrome/userChrome.css`

I hope this helps.

**Comment by owzim** [comment]
I am wondering, will a fix in the webapp manager be even ported to 22.0, or will users be stuck with this broken state unless they upgrade to 22.2 or 22.1 or wherever this gets fixed.?

**Comment by Mark3xtrm** [comment]
So no update on this? Mint team is veeeeeery slow to response

**Comment by mtwebster** [comment]
> I am wondering, will a fix in the webapp manager be even ported to 22.0, or will users be stuck with this broken state unless they upgrade to 22.2 or 22.1 or wherever this gets fixed.?

Once a fix is determined, this is something that *will* probably be backported. However, repairing existing user web apps may require the user to make manual edits.

> So no update on this? Mint team is veeeeeery slow to response

This hasn't been ignored. We're playing catch up with these changes just like everyone else. The behavior has changed even from 133.0 to 133.0.3 (yesterday).

**Comment by NicTanghe** [comment]
My userchrome.css also broke.
Normally everything is hidden unless i hit the top (like in f11 mode)
but now the url and its little box with the shield and the lock is always visable no matter what. and moving mouse upwards doesn`t make anything appear.

**Comment by clefebvre** [comment]
This messes with popups.. you can try the following URL

https://www.linuxmint.com/matrix.php

See the popup which asks to allow persistent storage at the top, blinking in and out.

**Comment by bassmannate** [comment]
> this is what i did for now for every `chrome/userChrome.css` under my apps in `.local/share/ice/firefox/`:
> 
> #navigator-toolbox {
>     border-bottom: none !important;
> }
> 
> #nav-bar, #identity-box, #tabbrowser-tabs, #TabsToolbar, #nav-bar * {
>     visibility: collapse !important;
> }
> (you can do it once and copy the file around of course, or use some bash magic)
> 
> it will always hide the bars but i never need them in my web apps.
> 
> i also (hot-patched) it for future web apps under `/usr/share/webapp-manager/firefox/profile/chrome/userChrome.css`
> 
> it works beautifully for my needs.
> 
> ![image](https://github.com/user-attachments/assets/eb60958c-9ac2-4bf3-b704-06d3a37f9413)

Just wanted to say that here we are in January 2026 and a similar issue is turning up. Hovering over the top of a webapp window causes the URL bar to pop-over content. Adding this to the CSS fixes it. Will require a restart of the webapp in question.
